From: Matus Kral <matuskral@me.com>
Date: Wed, 30 Jan 2013 05:11:12 +0100
Subject: solves dependencies problem, endless loop
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

I suppose this needs fixing in mountall original code, but for the time being.

specific case like this:

fs                   mountpoint
zfs                  /zfs
zfs/home             /home
zfs/home/1           /zfs/test/1
zfs/test             /zfs/test

mountall ended up in a endless loop, waiting for dependencies. if mounts are read ordered by mount points, it's ok. 
---
 src/mountall.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mountall.c b/src/mountall.c
index 2be5903..5789d7e 100644
--- a/src/mountall.c
+++ b/src/mountall.c
@@ -762,7 +762,7 @@ parse_zfs_list (void)
 	size_t bufsz = 4096;
 	FILE *zfs_stream;
 	const char zfs_command[] =
-	  "/sbin/zfs list -H -t filesystem -o name,mountpoint,canmount";
+	  "/sbin/zfs list -H -t filesystem -s mountpoint -o name,mountpoint,canmount";
 	int zfs_result;
 
 	nih_debug ("parsing ZFS list");
-- 
